### PR TITLE
Submatrix for other Matrix Types

### PR DESCRIPTION
**Pull request by Marvin-Beckmann**
**Description**

This PR implements the window functionality for all other matrix types. These include
- [x] MatQ
- [x] MatZq
- [x] MatPolyOverZ
- [x] MatPolynomialRingZq

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/integer_mod_q/mat_polynomial_ring_zq/get.rs [hunk 1]
```diff
@@ -158,6 +158,54 @@ impl GetEntry<PolynomialRingZq> for MatPolynomialRingZq {
...
+    pub fn get_submatrix(
+        &self,
+        row1: impl TryInto<i64> + Display,
+        row2: impl TryInto<i64> + Display,
+        col1: impl TryInto<i64> + Display,
+        col2: impl TryInto<i64> + Display,
+    ) -> Result<Self, MathError> {
+        Ok(MatPolynomialRingZq {
+            matrix: self.matrix.get_submatrix(row1, row2, col1, col2)?,
+            modulus: self.modulus.clone(),
```
Comment (Marcel583) [inline src/integer_mod_q/mat_polynomial_ring_zq/get.rs:206]: Why not self.get_mod()? I think both do the same.

File: src/integer_mod_q/mat_polynomial_ring_zq/get.rs [hunk 2]
```diff
@@ -158,6 +158,54 @@ impl GetEntry<PolynomialRingZq> for MatPolynomialRingZq {
 }
 
 impl MatPolynomialRingZq {
+    /// Returns a deep copy of the submatrix defined by the given parameters.
+    /// All entries starting from `(row1, col1)` to `(row2, col2)`(inclusively) are collected in
+    /// a new matrix.
+    /// Note that `row1 >= row2` and `col1 >= col2` must hold. Otherwise the function will panic.
+    ///
```
Comment (Marcel583) [inline src/integer_mod_q/mat_polynomial_ring_zq/get.rs:165]: Maybe implement get_row and get_column if it is not there.

**Review by Marcel583 — CHANGES_REQUESTED** [review]

File: src/integer_mod_q/mat_polynomial_ring_zq/get.rs (hunk 2 above)
Comment (Marvin-Beckmann) [reply src/integer_mod_q/mat_polynomial_ring_zq/get.rs:165]: will make a separate PR for it

File: src/integer_mod_q/mat_polynomial_ring_zq/get.rs (hunk 1 above)
Comment (Marvin-Beckmann) [reply src/integer_mod_q/mat_polynomial_ring_zq/get.rs]: done

**Pull request merged**